src/main/java/com/accantosystems/stratoss/vnfmdriver/driver/GrantDriver.java

```diff
@@ -73,7 +79,10 @@ public GrantCreationResponse requestGrant(GrantRequest grantRequest) throws Gran
         final HttpEntity<GrantRequest> requestEntity = new HttpEntity<>(grantRequest, headers);
         final ResponseEntity<Grant> responseEntity;
         try {
+            UUID uuid = UUID.randomUUID();
+            LoggingUtils.logEnabledMDC(grantRequest != null ? grantRequest.toString() : null, MessageType.REQUEST, MessageDirection.RECEIVED, uuid.toString(),MediaType.APPLICATION_JSON.toString(), "https",getRequestProtocolMetaData(url) ,grantRequest.getVnfLcmOpOccId());
             responseEntity = authenticatedRestTemplate.exchange(url, HttpMethod.POST, requestEntity, Grant.class);
+            LoggingUtils.logEnabledMDC(responseEntity.getBody() != null ? responseEntity.getBody().toString() : null, MessageType.RESPONSE,MessageDirection.SENT,uuid.toString(),MediaType.APPLICATION_JSON.toString(), "https",getProtocolMetaData(url,responseEntity),grantRequest.getVnfLcmOpOccId());
```

Comment:
Instead of checking null for the responseEntity.getBody(), check on the responseEntity only.